unit/testing-utils/require_parse_tree.cpp

```diff
@@ -12,44 +12,42 @@
 /// lambda_method_handle_map with a matching descriptor. Will fail if no
 /// matching lambda entry found.
 /// \param parsed_class: the class to inspect
-/// \param descriptor: the descriptor the lambda method should have
-/// \param entry_index: the number to skip (i.e. if multiple entries with the
-/// same descriptor
+/// \param lambda_method_ref: the reference/descriptor of the lambda method
+/// to which this lambda entry points to, must be unique
+/// \param method_type: the descriptor the lambda method should have
 /// \return
 require_parse_tree::lambda_method_handlet
 require_parse_tree::require_lambda_entry_for_descriptor(
   const java_bytecode_parse_treet::classt &parsed_class,
-  const std::string &descriptor,
-  const size_t entry_index)
+  const std::string &lambda_method_ref,
+  const std::string &method_type)
 {
-  REQUIRE(entry_index < parsed_class.lambda_method_handle_map.size());
   typedef java_bytecode_parse_treet::classt::lambda_method_handle_mapt::
     value_type lambda_method_entryt;
 
-  size_t matches_found = 0;
+  INFO(
+    "Looking for entry with lambda_method_ref: " << lambda_method_ref
+                                                 << " and method_type: "
+                                                 << method_type);
+  const auto num_matches = std::count_if(
+    parsed_class.lambda_method_handle_map.begin(),
+    parsed_class.lambda_method_handle_map.end(),
+    [&method_type, &lambda_method_ref](const lambda_method_entryt &entry) {
```

Comment:
Perhaps store this predicate as a `is_match` or remove the second loop by using `copy_if` and checking there is exactly one (bit clunky I know)